moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h

```diff
@@ -69,13 +69,30 @@ class ActionBasedControllerHandle : public ActionBasedControllerHandleBase
 {
 public:
   ActionBasedControllerHandle(const std::string& name, const std::string& ns)
-    : ActionBasedControllerHandleBase(name), namespace_(ns), done_(true)
+    : ActionBasedControllerHandleBase(name), namespace_(ns), done_(true), nh_("~")
   {
     controller_action_client_.reset(new actionlib::SimpleActionClient<T>(getActionName(), true));
     unsigned int attempts = 0;
-    while (ros::ok() && !controller_action_client_->waitForServer(ros::Duration(5.0)) && ++attempts < 3)
-      ROS_INFO_STREAM_NAMED("moveit_simple_controller_manager", "Waiting for " << getActionName() << " to come up");
+    double timeout;
+    nh_.param("trajectory_execution/controller_connection_timeout", timeout, 5.0);
 
+    if (timeout == 0.0)
+    {
+      ROS_WARN_STREAM_NAMED("moveit_simple_controller_manager", "No controller specified. Waiting indefinitely...");
```

Comment:
I believe "No controller specified" is wrong and confusing here.
I would propose to remove the warning here - also because it will be triggered *every time* even if the controller is already connected.

Reply:
Well it is only printed once and it is just to warn that it is going to wait indefinitely for the controller to connect. Should I just improve the message to "No controller connected. Waiting indefinitely..." or just remove it?